spec/requests/movies_spec.rb

```diff
@@ -0,0 +1,337 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe '/movies', type: :request do
+  let(:user) { create :user }
+  let(:manager) { create :user, email: 'manager@gmail.com', role: :manager }
+  describe 'GET /movies' do
+    subject(:request) { get movies_url }
+
+    before { request }
+
+    it 'returns successful response' do
+      expect(response).to be_successful
+    end
+
+    it 'renders proper template' do
+      expect(response).to render_template('movies/index')
+    end
+  end
+
+  describe 'GET /movies/movie_id' do
+    let(:movie) { create :movie }
+    it 'returns successful response' do
+      get("/movies/#{movie.id}")
+      expect(response.status).to eq(200)
+    end
+
+    it 'renders proper template' do
+      get("/movies/#{movie.id}")
+      expect(response).to render_template('movies/show')
+    end
+  end
+
+  describe 'GET /movies/new' do
+    context 'when user not signed in' do
+      it 'redirects to sign_in' do
+        get('/movies/new')
+        expect(response).to redirect_to('/users/sign_in')
+      end
+
+      it 'returns redirect status' do
+        get('/movies/new')
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user without permission' do
+      before { sign_in user }
+      it 'redirects to root' do
+        get('/movies/new')
+        expect(response).to redirect_to('/')
+      end
+
+      it 'returns redirect status' do
+        get('/movies/new')
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context 'when user with permission' do
+      before { sign_in manager }
+      it 'returns successful response' do
+        get('/movies/new')
+        expect(response.status).to eq(200)
+      end
+
+      it 'renders proper template' do
+        get('/movies/new')
+        expect(response).to render_template('movies/new')
+      end
+    end
+  end
+
+  describe 'POST /movies' do
+    let(:params) do
+      {
+        movie: {
+          title:,
+          description: Faker::Movie.quote,
+          duration:
+        }
+      }
+    end
```

Comment:
FactoryBot has a method for generating params hash
```suggestion
    let(:params) { attributes_for(:movie, title:, duration:) }
```